### PR TITLE
Remove parameters option of .likelihood() functions

### PR DESCRIPTION
**Pull request by adonath**
This PR removes the unused `parameters` option of the `Dataset.likelihood()` methods we currently have.

### CODE REVIEW

File: gammapy/utils/fitting/tests/test_fit.py [hunk 1]
```diff
@@ -10,27 +10,29 @@
 pytest.importorskip("iminuit")
 
 
-class MyModel(Model):
-    """Dummy model class."""
-
+class MyDataset:
```
Comment (cdeil) [inline gammapy/utils/fitting/tests/test_fit.py]: Derive from the `Dataset` base class?

File: gammapy/utils/fitting/tests/test_iminuit.py [hunk 2]
```diff
@@ -7,17 +7,19 @@
...
-    y = parameters["y"].value
-    z = parameters["z"].value
-    x_opt, y_opt, z_opt = 2, 3e5, 4e-5
-    x_err, y_err, z_err = 0.2, 3e4, 4e-6
-    return (
-        ((x - x_opt) / x_err) ** 2
-        + ((y - y_opt) / y_err) ** 2
-        + ((z - z_opt) / z_err) ** 2
-    )
+class MyDataset:
```
Comment (cdeil) [inline gammapy/utils/fitting/tests/test_iminuit.py]: Derive from the `Dataset` base class?
(Is this a duplicated test case? Can one variant be removed?)

File: gammapy/utils/fitting/tests/test_scipy.py [hunk 3]
```diff
@@ -4,17 +4,19 @@
...
-    y = parameters["y"].value
-    z = parameters["z"].value
-    x_opt, y_opt, z_opt = 2, 3e5, 4e-5
-    x_err, y_err, z_err = 0.2, 3e4, 4e-6
-    return (
-        ((x - x_opt) / x_err) ** 2
-        + ((y - y_opt) / y_err) ** 2
-        + ((z - z_opt) / z_err) ** 2
-    )
+class MyDataset:
```
Comment (cdeil) [inline gammapy/utils/fitting/tests/test_scipy.py]: Again?

I think the fitting backend unit tests should be re-structured, especially for iminuit, no?

Open a reminder issue for v0.13?

File: gammapy/utils/fitting/tests/test_sherpa.py [hunk 4]
```diff
@@ -6,17 +6,19 @@
...
-    y = parameters["y"].value
-    z = parameters["z"].value
-    x_opt, y_opt, z_opt = 2, 3e5, 4e-5
-    x_err, y_err, z_err = 0.2, 3e4, 4e-6
-    return (
-        ((x - x_opt) / x_err) ** 2
-        + ((y - y_opt) / y_err) ** 2
-        + ((z - z_opt) / z_err) ** 2
-    )
+class MyDataset:
```
Comment (cdeil) [inline gammapy/utils/fitting/tests/test_sherpa.py]: Fourth copy of the same testcase?

File: gammapy/utils/fitting/tests/test_iminuit.py (hunk 2 above)
Comment (adonath) [reply gammapy/utils/fitting/tests/test_iminuit.py]: Yes, this is partly a duplicated test case. The fitting backends are tested individually and via the `Fit` API again in `utils/fitting/test_fit.py`. I agree this should be refactored, I'll make a reminder issue for v0.13.

**Pull request merged**